File: app/src/main/java/me/ykrank/s1next/view/internal/LoadingViewModelBindingDelegateQuickSidebarImpl.java
package me.ykrank.s1next.view.internal;

import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import me.ykrank.s1next.databinding.FragmentBaseWithQuickSideBarBinding;
import me.ykrank.s1next.viewmodel.LoadingViewModel;

public final class LoadingViewModelBindingDelegateQuickSidebarImpl
        implements LoadingViewModelBindingDelegate {

    private final FragmentBaseWithQuickSideBarBinding binding;

    public LoadingViewModelBindingDelegateQuickSidebarImpl(
            FragmentBaseWithQuickSideBarBinding binding) {
        this.binding = binding;
    }

    @Override
    public View getRootView() {
        return binding.getRoot();
    }

    @Override
    public SwipeRefreshLayout getSwipeRefreshLayout() {
        return binding.swipeRefreshLayout;
    }

    @Override
    public RecyclerView getRecyclerView() {
        return binding.recyclerView;
    }

    @Override
    public void setLoadingViewModel(LoadingViewModel loadingViewModel) {
        binding.setLoadingViewModel(loadingViewModel);
    }
}

File: app/src/main/java/me/ykrank/s1next/view/internal/LoadingViewModelBindingDelegateQuickSidebarImpl.kt
package me.ykrank.s1next.view.internal

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import com.github.ykrank.androidtools.ui.internal.LoadingViewModelBindingDelegate
import com.github.ykrank.androidtools.ui.vm.LoadingViewModel
import me.ykrank.s1next.databinding.FragmentBaseWithQuickSideBarBinding

class LoadingViewModelBindingDelegateQuickSidebarImpl<D>(
    private val binding: FragmentBaseWithQuickSideBarBinding
) : LoadingViewModelBindingDelegate<D> {
    override val rootView: View
        get() = binding.root
    override val swipeRefreshLayout: SwipeRefreshLayout
        get() = binding.swipeRefreshLayout
    override val recyclerView: RecyclerView
        get() = binding.recyclerView

    override fun setLoadingViewModel(loadingViewModel: LoadingViewModel<D>) {
        binding.setLoadingViewModel(loadingViewModel)
    }
}
